implementations/micrometer-registry-otlp/src/main/java/io/micrometer/registry/otlp/OtlpMeterRegistry.java

```diff
@@ -0,0 +1,301 @@
+            SummaryDataPoint.Builder summaryData = SummaryDataPoint.newBuilder()
+                    .addAllAttributes(getTagsForId(histogramSupport.getId()))
+                    .setTimeUnixNano(wallTimeNanos)
+                    .setSum(histogramSnapshot.total())
+                    .setCount(histogramSnapshot.count());
+            for (ValueAtPercentile percentile : histogramSnapshot.percentileValues()) {
+                summaryData.addQuantileValues(SummaryDataPoint.ValueAtQuantile.newBuilder()
+                        .setQuantile(percentile.percentile()).setValue(percentile.value()));
+            }
+            metricBuilder.setSummary(Summary.newBuilder().addDataPoints(summaryData));
+            return metricBuilder.build();
+        }
+
+        return metricBuilder.setHistogram(Histogram.newBuilder()
+                .setAggregationTemporality(AggregationTemporality.AGGREGATION_TEMPORALITY_CUMULATIVE)
+                .addDataPoints(histogramDataPoint)).build();
+    }
+
+    private Metric writeFunctionTimer(FunctionTimer functionTimer) {
+        return getMetricBuilder(functionTimer.getId())
+                .setHistogram(Histogram.newBuilder()
+                        .addDataPoints(HistogramDataPoint.newBuilder()
+                                .addAllAttributes(getTagsForId(functionTimer.getId()))
+                                // TODO start time
+                                .setTimeUnixNano(TimeUnit.MILLISECONDS.toNanos(this.clock.wallTime()))
+                                .setSum(functionTimer.totalTime(getBaseTimeUnit()))
+                                .setCount((long) functionTimer.count())))
+                .build();
+    }
+
+    private Metric.Builder getMetricBuilder(Meter.Id id) {
+        Metric.Builder builder = Metric.newBuilder()
+                .setName(getConventionName(id));
+        if (id.getBaseUnit() != null) {
+            builder.setUnit(id.getBaseUnit());
+        }
+        if (id.getDescription() != null) {
+            builder.setDescription(id.getDescription());
+        }
+        return builder;
+    }
+
+    private Iterable<? extends KeyValue> getTagsForId(Meter.Id id) {
+        return id.getTags().stream()
+                .map(tag -> KeyValue.newBuilder()
+                        .setKey(tag.getKey())
+                        .setValue(AnyValue.newBuilder().setStringValue(tag.getValue()).build())
+                        .build())
+                .collect(Collectors.toList());
+    }
+
+    private Iterable<KeyValue> getResourceAttributes() {
+        List<KeyValue> attributes = new ArrayList<>();
+        // TODO How to expose configuration of the service.name
+        attributes.add(KeyValue.newBuilder().setKey("service.name").setValue(AnyValue.newBuilder().setStringValue("unknown_service")).build());
+        attributes.add(KeyValue.newBuilder().setKey("telemetry.sdk.name").setValue(AnyValue.newBuilder().setStringValue("io.micrometer")).build());
```

Comment:
I wasn't sure what naming convention is best here. `io.micrometer` seemed reasonable as our reverse domain. Alternatively, maybe `micrometer.core` is better as it's our automatic module name for micrometer-core.
Documentation: https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/resource/semantic_conventions/README.md#telemetry-sdk

Reply:
How about simply `micrometer`?

Reply:
That's probably also fine. I was basing my suggestions off the wording in the docs:
> If another SDK, like a fork or a vendor-provided implementation, is used, this SDK MUST set the attribute `telemetry.sdk.name` to the fully-qualified class or module name of this SDK's main entry point or another suitable identifier depending on the language.

Also, maybe we should distinguish between micrometer-core and tracing.